src/sentry/tsdb/snuba.py

```diff
@@ -73,7 +148,8 @@ def get_data(
             groupby.append(model_aggregate)
             model_aggregate = None
 
-        keys_map = dict(zip(model_columns, self.flatten_keys(keys)))
+        columns = (model_query_settings.groupby, model_query_settings.aggregate)
+        keys_map = dict(zip(columns, self.flatten_keys(keys)))
```

Comment:
I think that should be fine, no? Cause `self.flatten_keys` has a defined behavior if the keys is a dictionary. Maybe I'm missing something.

I think doing something like this would also make it easier to understand but want to understand the problem better before I do that:
```
-        columns = (model_query_settings.groupby, model_query_settings.aggregate)
-        keys_map = dict(zip(columns, self.flatten_keys(keys)))
+        (group_by_values, aggregate_values) = self.flatten_keys(keys)
+        keys_map = {
+            model_query_settings.groupby: group_by_values,
+            model_query_settings.aggregate: aggregate_values,
+        }
```

Anyway, I'll deploy the PR now and depending on our discussion, will make a cleanup fix. :+1: